lxd/containers.go

```diff
@@ -64,28 +53,18 @@ func containersGet(d *Daemon, r *http.Request) Response {
 	return SyncResponse(true, result)
 }
 
-func containersPost(d *Daemon, r *http.Request) Response {
-	shared.Debugf("responding to create")
-
-	if d.id_map == nil {
-		return BadRequest(fmt.Errorf("shared's user has no subuids"))
-	}
-
-	req := containerPostReq{}
-	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
-		return BadRequest(err)
-	}
-
-	if req.Name == "" {
-		req.Name = strings.ToLower(petname.Generate(2, "-"))
-		shared.Debugf("no name provided, creating %s", req.Name)
-	}
+type containerImageSource struct {
+	Type string `json:"type"`
+	URL  string `json:"url"`
+	Name string `json:"name"`
+}
```

Comment:
I realize you've just moved that code, but if we're doing changes in that area, can we make that bit match the spec?

The valid list of source fields from rest-api.md is:
 - type (string)
 - alias (string)
 - fingerprint (string)
 - properties (map of string:string)
 - mode (string)
 - server (string)
 - secret (string)
 - operation (string)

I suspect the "name" field we used so far is going to be either "fingerprint" or "alias".